app/services/authorisation.rb

```diff
@@ -0,0 +1,21 @@
+class Authorisation
+  PERMISSION_METHOD_REGEXP = /^can_(\w+)\?$/.freeze
+
+  def initialize(actor:)
+    @actor = actor
+  end
+
+  def can_make_offer?(application_choice:)
+    @actor.is_a?(SupportUser) || @actor.providers.include?(application_choice.course.provider)
+  end
+
+  class AssertError < StandardError; end
+
+  instance_methods.select { |m| m.match PERMISSION_METHOD_REGEXP }.each do |method|
```

Comment:
This is very nifty! 🏎 

It's a bit hard to follow, though. Could we make do with a single `assert_can_make_offer?` method for now, and refactor when we have some duplication?